### PR TITLE
[browser][MT] unify pthread pool size to 32

### PR DESCRIPTION
**Pull request by pavelsavara**
- bump default thread pool size to 32 and unify to that default for all tests
 - see which existing active issues are caused by the same reason
 - keep ActiveIssue tracking it, because we know that it's too large default
     - https://github.com/dotnet/runtime/issues/91538

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.HostFactoryResolver/tests/HostFactoryResolverTests.cs [hunk 1]
```diff
@@ -279,7 +284,7 @@ public void ApplicationNameSetFromArgument()
 
         [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
         [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(NoSpecialEntryPointPattern.Program))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/91538", typeof(PlatformDetection), nameof(PlatformDetection.IsWasmThreadingSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/73420", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
Comment (ilonatommy) [inline src/libraries/Microsoft.Extensions.HostFactoryResolver/tests/HostFactoryResolverTests.cs:287]: This is a closed issue, is it a typo or should we reopen it?

File: src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs [hunk 2]
```diff
@@ -17,7 +17,9 @@ internal sealed partial class PortableThreadPool
 
         private const short MaxPossibleThreadCount = short.MaxValue;
 
-#if TARGET_64BIT
+#if TARGET_BROWSER
+        private const short DefaultMaxWorkerThreadCount = 16;
```
Comment (ilonatommy) [inline src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs:21]: not 32?

**Review by ilonatommy — COMMENTED** [review]
I checked and all removed issues are relevant

File: src/libraries/Microsoft.Extensions.HostFactoryResolver/tests/HostFactoryResolverTests.cs (hunk 1 above)
Comment (pavelsavara) [reply src/libraries/Microsoft.Extensions.HostFactoryResolver/tests/HostFactoryResolverTests.cs:287]: bad merge, sorry

File: src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs (hunk 2 above)
Comment (pavelsavara) [reply src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs:21]: there is 
- pthread pool (emscripten level)
- managed thread pool (portion of pthread pool )
- threads allocated by user code (will use the rest of the available pthreads, but even that could be not enough)

File: eng/pipelines/extra-platforms/runtime-extra-platforms-wasm.yml [hunk 3]
```diff
@@ -112,14 +112,14 @@ jobs:
...
+      extraBuildArgs: /p:MonoWasmBuildVariant=multithread /p:WasmEnableThreads=true /p:AotHostArchitecture=x64 /p:AotHostOS=$(_hostedOS)
       isExtraPlatformsBuild: ${{ parameters.isExtraPlatformsBuild }}
       isWasmOnlyBuild: ${{ parameters.isWasmOnlyBuild }}
       # Always run for runtime-wasm because tests are not run in runtime
       alwaysRun: ${{ parameters.isWasmOnlyBuild }}
 
-       # NOTE - Since threading is experimental, we don't want to block mainline work
-      shouldContinueOnError: true
+      # NOTE - Since threading is experimental, we don't want to block mainline work
+      # TODO put back shouldContinueOnError: true
```
Comment (pavelsavara) [inline eng/pipelines/extra-platforms/runtime-extra-platforms-wasm.yml:122]: we should put this back before merge

**Pull request closed without merging**